designer/src/main/java/org/restcomm/connect/rvd/commons/http/CustomHttpClientBuilder.java

```diff
@@ -17,63 +19,129 @@
 import org.restcomm.connect.rvd.RvdConfiguration;
 
 import javax.net.ssl.SSLContext;
-
+import org.apache.http.HttpHost;
+import org.apache.http.config.Registry;
+import org.apache.http.config.RegistryBuilder;
+import org.apache.http.conn.routing.HttpRoute;
+import org.apache.http.conn.socket.ConnectionSocketFactory;
+import org.apache.http.conn.socket.PlainConnectionSocketFactory;
+import org.apache.http.conn.util.PublicSuffixMatcher;
+import org.apache.http.conn.util.PublicSuffixMatcherLoader;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.restcomm.connect.rvd.configuration.RvdMaxPerHost;
 
 /**
- * Creates an HttpClient with the desired ssl behaviour according to configuration
+ * Creates an HttpClient with the desired ssl behaviour according to
+ * configuration
+ *
  * @author orestis.tsakiridis@telestax.com (Orestis Tsakiridis)
  *
  */
-
 public class CustomHttpClientBuilder {
-    private SslMode sslMode;
+
+    private RvdConfiguration configuration;
 
     public CustomHttpClientBuilder(RvdConfiguration configuration) {
-        this.sslMode = configuration.getSslMode();
+        this.configuration = configuration;
     }
 
     // returns an apache http client
-    public CloseableHttpClient buildHttpClient(Integer timeout) {
-        if ( sslMode == SslMode.strict ) {
-            return buildStrictClient(timeout);
-        }
-        else
-            return buildAllowallClient(timeout);
+    public CloseableHttpClient buildExternalHttpClient() {
+        return buildClient(configuration.getExternalServiceTimeout(),
+                configuration.getExternalServiceMaxConns(),
+                configuration.getExternalServiceMaxConnsPerRoute(),
+                configuration.getExternalServiceTTL(),
+                configuration.getExternalServiceMaxPerRoute());
     }
 
     public CloseableHttpClient buildHttpClient() {
-        return buildHttpClient(null);
+        return buildClient(configuration.getDefaultHttpTimeout(),
+                configuration.getDefaultHttpMaxConns(),
+                configuration.getDefaultHttpMaxConnsPerRoute(),
+                configuration.getDefaultHttpTTL(),
+                configuration.getDefaultHttpMaxPerRoute());
     }
 
-    private CloseableHttpClient buildStrictClient(Integer timeout) {
-        // set the timeout
+    private CloseableHttpClient buildClient(Integer timeout,
```

Comment:
A little worried on SSL regressions here. apache-client/SSL has traditionally been source of a lot of headaches both in RVD and restcomm. Since there is a lot of refactoring here, are you confident that the refactored stuff works in particular with actual https:// locations ? Did you do any manual testing ?

Reply:
nope, i didnt do any testing on SSL, but the code is mostly the same, but but in different places, and taking advantage of builder pattern, instead of calling the builder in different places.

Reply:
ok, np @jaimecasero. I'll try to test a coupld of cases (lke our cloud :-) to make sure everything is fine.